### PR TITLE
feat: added sequencer set polling and consensus msgs queue

### PR DESCRIPTION
**Pull request by keruch**
# PR Standards

ADR links 

1. https://www.notion.so/dymension/ADR-x-Sequencer-Reward-Address-On-Rollapp-da84af6888c141d0a4c1a8df256a5025
2. https://www.notion.so/dymension/ADR-Whitelisted-Relayer-Fee-Exemption-in-Rollapp-104a4a51f86a80e8ad94c916c87deb17

Closes 
https://github.com/dymensionxyz/dymension/issues/1248
https://github.com/dymensionxyz/dymension/issues/1226
https://github.com/dymensionxyz/dymint/issues/1153
https://github.com/dymensionxyz/dymint/issues/1125

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Comment by danwt** [comment]
linter

**Review by danwt — COMMENTED** [review]
Only did a quick scan, but might it be better to use an off shelf queue?
This is a ubiquitious data structure with numerous library implementations
e.g .https://github.com/liyue201/gostl?tab=readme-ov-file#queue

File: types/sequencer_set.go [hunk 1]
```diff
@@ -134,13 +197,36 @@ func (s *SequencerSet) GetByConsAddress(cons_addr []byte) *Sequencer {
 	return nil
 }
 
+// SequencerListRightOuterJoin returns a set of sequencers that are in B but not in A.
+// CONTRACT: both A and B do not have duplicates!
```
Comment (omritoptix) [inline types/sequencer_set.go:201]: not sure I follow this. why is this assumption necessary? worst case diff is empty no?

File: block/sequencers.go [hunk 2]
```diff
@@ -51,6 +50,43 @@ func (m *Manager) MonitorSequencerRotation(ctx context.Context, rotateC chan str
...
+		}
+	}
+}
+
+// HandleSequencerSetUpdate calculates the diff between hub's and current sequencer sets and
+// creates consensus messages for all new sequencers. The method updates the current state
+// and is not thread-safe. Returns errors on serialization issues.
+func (m *Manager) HandleSequencerSetUpdate(newSet []types.Sequencer) error {
+	// find new (updated) sequencers
+	newSequencers := types.SequencerListRightOuterJoin(m.State.Sequencers.Sequencers, newSet)
```
Comment (omritoptix) [inline block/sequencers.go:77]: what if newSequencers is empty? probably worth returning early.

**Review by omritoptix — COMMENTED** [review]
good job! a few comments/suggestions.

**Comment by omritoptix** [comment]
blocked by #1149

**Review by omritoptix — APPROVED** [review]

**Review by omritoptix — DISMISSED** [review]

**Review by omritoptix — APPROVED** [review]

**Review by omritoptix — APPROVED** [review]

**Pull request merged**